http/handler.go

```diff
@@ -273,6 +273,11 @@ func handleUIHeaders(core *vault.Core, h http.Handler) http.Handler {
 
 func handleUI(h http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+
+		// The fileserver handler strips trailing slashes and does a redirect.
+		// We don't want the redirect to happen so we preemptively trim the slash
+		// here.
+		req.URL.Path = strings.TrimSuffix(req.URL.Path, "/")
```

Comment:
I have a couple of question:

1. Do we need to worry about query params? e.g. "example.com/a/?q=foo" -> "example.com/a?q=foo"
2. A URL struct has a number of elements, including RawPath and some other parsed stuff. I don't know when they get populated and how they're used, but I wonder about consistency. I do see that the http.StripPrefix function, which is try to do something similar, has a more elaborate process to change the path. Not sure if this is strictly required or not. (https://github.com/golang/go/blob/master/src/net/http/server.go#L1990)

Reply:
I also manually tested queryParams (the kv UI supports filtering the list) - http://localhost:8200/ui/vault/secrets/secret/list/a/?pageFilter=b refreshed and worked as expected.

Reply:
After some more review I think point 2 is not an issue here.

Reply:
1) the `Path` variable doesn't include query params, so this shouldn't be an issue
2) I don't think that's strictly necessary, really we just don't want the fileserver to view the path as a directory since we are re-writing it to a file (`index.html`) elsewhere.